src/sagemaker/model.py

```diff
@@ -302,6 +302,16 @@ def transformer(self, instance_count, instance_type, strategy=None, assemble_wit
                            env=env, tags=tags, base_transform_job_name=self.name,
                            volume_kms_key=volume_kms_key, sagemaker_session=self.sagemaker_session)
 
+    def delete_model(self):
+        """Delete an Amazon SageMaker ``Model``.
+
+        Raises: ValueError if model is not deployed yet.
+
+        """
+        if self.name is None:
+            raise ValueError('The SageMaker model is not deployed yet.')
```

Comment:
For the exception, I would specify something along the lines of...

"The SageMaker model must be deployed first before attempting to delete."

Although, the small problem here is that technically my SageMaker model can exist before I actually create an endpoint.

Reply:
I think this None check can also go in the session class. That way you don't have to check in the Transformer class either.

The unit test for this is missing I believe.